File: database/transaction/transaction-template/src/main/java/com/example/ApplicationEventListener.java
package com.example;

import static java.util.stream.Collectors.toList;

import java.util.List;
import java.util.stream.IntStream;
import java.util.stream.StreamSupport;

import org.springframework.boot.context.event.ApplicationReadyEvent;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Component
@RequiredArgsConstructor
@Slf4j
public class ApplicationEventListener {
    private final UserService userService;
    private final UserRepository userRepository;

    @EventListener(ApplicationReadyEvent.class)
    public void readyEvent() {
        IntStream.rangeClosed(1, 3)
                 .forEach(i -> {
                     User user = new User();
                     user.setName("name" + i);
                     log.info("{}", userService.create(user));
                 });

        List<User> users = StreamSupport.stream(userRepository.findAll().spliterator(), false)
                                        .collect(toList());
        users.forEach(user -> log.info("{}", user));

        User user = users.get(0);
        user.setName("name1-1");
        userService.update(user);

        userService.delete(users.get(1).getId());

        userRepository.findAll()
                      .forEach(entity -> log.info("{}", entity));
    }
}

File: rdb/exposed/src/main/kotlin/com/example/ApplicationEventListener.kt
package com.example

import org.springframework.boot.context.event.ApplicationReadyEvent
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component
import org.springframework.transaction.support.TransactionTemplate

@Component
class ApplicationEventListener(
    private val userService: UserService,
    private val transactionTemplate: TransactionTemplate
) {
    @EventListener(ApplicationReadyEvent::class)
    fun readyEvent() {
        val users = (1..5).map { User(0, "name$it") }.toList()
        userService.insert(users)
        userService.selectAll().forEach { println(it) }

        transactionTemplate.execute {
            val user = userService.select(1)
            userService.update(User(user.id, user.name + "__"))
            userService.delete(2)
        }
        userService.selectAll().forEach { println(it) }
    }
}
